### DISCUSSION TITLE
Customisation questions

### DISCUSSION THREAD

**Discussion by Zabooby** [Astromations/FlyingFox#1] [General]
Hi, I'm currently using this [css snippet](https://github.com/piroor/treestyletab/wiki/Code-snippets-for-custom-style-rules#full-auto-showhide-theme) (including the TST settings) from the TST wiki to autohide the sidebar, I tries your css by placing;
```
:root.left tab-item .sound-button:after {
    transform: translate(30px, -5px); !important;
    transition: all calc(var(--animation-duration)*2.2) var(--ease-out) !important;
}
```
in TST settings and;
```
:root.left tab-item:hover .sound-button:after {
  transform: translateX(-25px);
}
```
based on your screenshots, but it is broken (I'm guessing because of conflicts with the snippet above) any ideas?
![image](https://user-images.githubusercontent.com/78969986/233720717-56e42c73-2608-4eb4-877f-7c2c7cb41ee5.png)
![image](https://user-images.githubusercontent.com/78969986/233720754-721267d0-1181-4b31-830a-6aeed2de261e.png)

Also while using the snippet above, the animation while expanding is smooth but when collapsing just sidebar snaps in. FlyingFox has got buttery smooth animations both when collapsing and expanding, what parts of the css can I use to get the same effect? 

P.S. Thanks for reviving FlyingFox!

**Comment by Astromations** [comment]
The code is just meant to make the "playing/muted" icon hover on top of the favicon of the tab. If your goal is to make the playing icon disappear entirely, just set display to none. 

```
tab-item .sound-button {
    display: none;
}
```

As for the snippet, the problem is that Firefox is struggling to re-proportion the entire window as it animates the width of the sidebar. Firefox seems to have no trouble compressing the window but it finds it hard expanding it back out to fill in the space left by the sidebar as it collapses. It was only when I set the transition time to 4 seconds that Firefox was able to keep up. I don't think its an issue that can be solved by tweaking the css. You'd need a different method to collapse the sidebar entirely.

 I'd recommend just using the one that comes with FlyingFox, that being userChrome-hover. It works by just shoving the sidebar off screen until on the icons show, so that way, Firefox doesn't have to reanimate the whole window, its just a matter of covering and uncovering content. Mind you, it requires the icons to be on the right side, which may or may not be a deal breaker for you 
 (but if its not, `:root.left tab-item tab-favicon { order: 10 !important; }`).
 
All the variables can be customized with root:
```
:root {
    --sidebar-width: 250px;
    --sidebar-collapsed-width: 50px;
    --transition-time: 0.2s;
    --ease-out: cubic-bezier(.22,.08,.63,.74)
    }
```
Just put that anywhere in your userChrome and tweak as you see fit.
Here's the content of userChrome-hover:
```
#sidebar-box {
    z-index: 1000 !important;
    position: relative !important;
    min-width: var(--sidebar-collapsed-width) !important;
    max-width: var(--sidebar-collapsed-width) !important;
}

#sidebar-header,
#sidebar-splitter {
    display: none !important;
}

/* inner width = expanded width & move inner to only show initial-width long section */
#sidebar-box #sidebar {
    position: absolute !important;
    min-width: var(--sidebar-width) !important;
    max-width: var(--sidebar-width) !important;
    width: var(--sidebar-width) !important;
    transform: translateX(calc(var(--sidebar-collapsed-width) - var(--sidebar-width)));
    transition: all var(--transition-time) var(--ease-out);
}

/* move inner to show entire sidebar */
#sidebar-box #sidebar:hover {
    transform: translateX(0px) !important;
    /* box-shadow: 2px 0 33px -3px var(--sidebar-shadow-color); */
}

#sidebar-box[sidebarcommand="viewBookmarksSidebar"] #sidebar {
    transform: translateX(0px) !important;
    box-shadow: 2px 0 33px -3px var(--sidebar-shadow-color);
}
/* hide sidebar header for tree style tabs sidebar */
#sidebar-box[sidebarcommand="treestyletab_piro_sakura_ne_jp-sidebar-action"] #sidebar-header {
    display: none;
}

toolbar {
    border: none !important;
}

/* Hide splitter, when using Tree Style Tab. */
#sidebar-splitter {
    display: none !important;
}
```
Hope that helps!

**Comment by Zabooby** [comment]
Thanks for the detailed response, after some tweaking, it's clear that that getting the result I want would take too much work.

 I'll just use your fork and customise it instead. Here are some more questions and some small bugs I found: 
- With the css I linked above pinned tabs would be animated and fold on each other, (removing the following empty space that is shown when tabs are pinned). 
![image](https://user-images.githubusercontent.com/78969986/233777557-5026cbc8-af6a-448a-a8ee-55f7e1222c77.png)

Would you be interested in implementing this into FF or are there issues that could arise from it? 
 

https://user-images.githubusercontent.com/78969986/233316857-bc26e7ce-3c22-4541-a0d6-706e5378fc11.mp4
- Also when a pinned tab is active the background around it is slightly misaligned, the hover effect is also misaligned
- ![image](https://user-images.githubusercontent.com/78969986/233773869-b3b7fd56-c3a9-402b-8391-d9bc9c74a417.png)

- How do I hide the scrollbar?
- All right click menus seem to be bigger and more spaced out than normal how do I reset their size? 
- There seems to be a lot of rounding in various parts of the ui, how do I reset rounding for right click and drop-down menus, the url menu and this:
- ![image](https://user-images.githubusercontent.com/78969986/233774302-ad911f8e-80bc-4f58-9972-6bfd75b6f0b2.png)

- And finally, how do I reduce the spacing in the sidebar, indicated by the areas in red;
- ![image](https://user-images.githubusercontent.com/78969986/233776776-24bf0de4-a8b7-4eb0-8f15-7c2a31652d86.png)
- ![image](https://user-images.githubusercontent.com/78969986/233776848-50627440-97ea-46c3-8ae4-af056fc196d8.png)

**Reply by Astromations** [reply]
That's quite the list. 

1. I've fixed the alignment for the pinned tabs icons. I was also working on adding some extra padding between them but that's proving to be quite the hassle. Anyway, I'll be pushing that commit not too long from now so look out for that.

2. I really like the idea of the pinned tabs moving back into place when you stop hovering, unfortunately, the snippet you're using collapses the sidebar in a fundamentally different way than FF, so I'd need to come up with a new way to do that if it's even possible. 

3. The scrollbar can be hidden with the following:
```
#tabbar.overflow {
  -ms-overflow-style: none;  /* IE and Edge */
  scrollbar-width: none; 
}
```
Just paste that in your treestyletabs' css and it should be gone.

4. Regarding the new menu width, its was a rather opinionated change. I get you and others might prefer to the old design so I'm going to make it an editable value in `variables.css`. Default is gonna be 10px and 3px so you can turn them down or up until you find a width that you like.
![image](https://user-images.githubusercontent.com/80211195/233793351-e4d6c2b4-9199-4db7-8fca-915c36aa1468.png)

5. Regrettably, rounding isn't actually all covered under one variable. For the individual elements you've mentioned: 
- Urlbar rounding can be found in line 144 of urlbar.css:
![image](https://user-images.githubusercontent.com/80211195/233793592-af9de34a-380c-495b-bbea-f4efa6da94a3.png)

- Popup rounding can be found on line 52 of popup.css:
![image](https://user-images.githubusercontent.com/80211195/233793625-d6ed2470-3041-427c-87f5-53db5b441bdc.png)

- And the window rounding can be found on line 18 of navbar.css:
![image](https://user-images.githubusercontent.com/80211195/233793788-4da5220b-b1b7-45a7-a038-dbf63a9b1edc.png)

5. I'm not actually sure why that spacing appears for you. For me, tabs pretty much hug the left side of the sidebar. Make sure you've not only installed the treestyletab css but also the config file. You can do that by importing  it under the Development section. As for the padding on the left, just edit the `--distance-from-edge `variable in the treestyletabs.css editor. It's on line 47:
![image](https://user-images.githubusercontent.com/80211195/233793964-bd4aabaf-15c9-4275-ac00-cf7cdf1eb017.png)

Any other questions?

**Comment by Zabooby** [comment]
Thanks for all the solutions! 

For the spacing issue I tried using the .json file but I still get the issue, I also haven't tweaked any settings other than the ones mentioned above in the chome folder. I will try a fresh profile tomorrow and report back. 

Two more questions: 

Is the url bar the only menu that has a visible border? 
This is ~~probably~~ super nitpicky but I found the only parts of the ui not matching the FlyingFox theme is tooltips and the footer.

**Reply by Astromations** [reply]
Yeah the urlbar is the only menu with a border. It's easy to remove though. Just remove like 203 of urlbar.css
![image](https://user-images.githubusercontent.com/80211195/233823434-b4af1743-8383-4adc-ad4a-840ff19d9351.png)

I'll look into making tooltips integrate better with the rest of the theme. Can't say I know what you mean by footer though.

**Comment by Zabooby** [comment]
I might be using the wrong term but it's the little box that appears in the bottom left corner when hovering over links:
![image](https://user-images.githubusercontent.com/78969986/233825475-7b553eb2-d67e-437f-a2ec-a90caf4bf7d0.png)

**Reply by Astromations** [reply]
Ahhh. I'll see what I can do. I've already drawn up some css to update the tooltips. What do you think?
![image](https://user-images.githubusercontent.com/80211195/233825723-f6eda55f-6851-449f-8e2a-0228c0a633ab.png)

**Reply by Astromations** [reply]
Mind you, my colors are a bit different. I edited my config.css for a more muted color scheme.

**Reply by Zabooby** [reply]
Looks great!

**Reply by Astromations** [reply]
Thanks for the feedback, pushing the commit to interfaces.css right now.

**Comment by Zabooby** [comment]
Ok I made a fresh profile and I found the spacing issue is happening when switching the sidebar position from right to left.

https://user-images.githubusercontent.com/78969986/233829062-c83c5ca2-06ab-44da-838a-164b1538cd46.mp4

**Reply by Astromations** [reply]
Is the issue that there is too much or too little space?

**Reply by Zabooby** [reply]
Too much, I want the left side to have the same spacing as the right side.

**Reply by Astromations** [reply]
![image](https://user-images.githubusercontent.com/80211195/233834930-973177ee-1ddf-4e92-b3ae-fcacccf41179.png)
set `--distance-from-edge` to 2px

**Reply by Zabooby** [reply]
In the readme in the tst folder it says to have right side selected for "Style of contents for the sidebar position". Is this true even if the sidebar is on the left, 

In the video above the side bar has icons on the wrong side when using the right side setting. 

I tried setting `--distance-from-edge` to 2px while using the left side option but it has no effect.

**Reply by Astromations** [reply]
I'm yet to update the readme files. I'll fix that but for now they're just switched. 
Go to this line in your treestyletab css and make sure the second and last pixel settings are identical:
![image](https://user-images.githubusercontent.com/80211195/233839827-6709024f-6d95-446e-81fe-efff6496a2ca.png)

**Reply by Zabooby** [reply]
Instead of trying to change margins of the actual tab I think the padding before the text is what I'm actually after
![image](https://user-images.githubusercontent.com/78969986/233840937-c28b6d07-4d3d-46cc-8f9e-769af3694dd2.png)

Also where can I change the collapsed sidebar width?

**Reply by Astromations** [reply]
That can be changed here:
![image](https://user-images.githubusercontent.com/80211195/233841291-4b2c982b-9c8d-48a5-8ead-1a25fec16eeb.png)
You can also add a negative `margin-left` to push it further.

**Reply by Astromations** [reply]
Sidebar width can be changed in `config.css`
![image](https://user-images.githubusercontent.com/80211195/233841354-c4687447-cebb-463b-a820-1c8f36724456.png)

**Comment by Zabooby** [comment]
Ok no more questions (for now lol) I just have to tweak colours and everything will be set up. 

Thanks for being so patient and answering all my questions!

**Comment by Zabooby** [comment]
@Astromations after some more tweaking I found two more things I couldn't figure out how to change: 

- How do I remove dimming in the bookmarks panel
- How do I shrink the top bar, I've tried `browser.compactmode.show` but it doesn't seem to have any effect?

![image](https://user-images.githubusercontent.com/78969986/235511122-335fb08e-93ea-41fe-93b9-58927b2b24e4.png)

I also opened an issue (#2) related to bookmarks/history in the sidebar a few days ago. 

Other than that I've been enjoying FlyingFox a lot!

**Comment by Astromations** [comment]
1. Remove this bit of code on line 137 of navbar.css
![image](https://user-images.githubusercontent.com/80211195/235513133-d52d9322-9a43-468c-88db-e2484a3e906d.png)
2. Change line 7 to the property `height` instead of `min-height`. From there, you can change the height and fiddle with the padding and urlbar height (line 163 urlbar.css) until you find something you like.
![image](https://user-images.githubusercontent.com/80211195/235514026-1a653798-03da-4424-8c41-4c969ffc301e.png)

Been a bit busy so I missed your issue. I'll take a look.

**Reply by Zabooby** [reply]
No problem take as long as you need, it's only a minor issue anyway. 

Thanks for the solutions as always!

**Comment by Zabooby** [comment]
@Astromations Some quick questions: 
- How do I change the colour of the loading dot? 
- Although the splitter in the sidebar is set to be hidden it is still visible for me, similarly I have set the border for the status panel to be zero but it's still seen.  
   - `border: 0px solid var(--toolbar-field-focus-border-color) !important;`
![image](https://user-images.githubusercontent.com/78969986/236486946-fe1c6843-2e46-4756-adc3-9a967f0a9fcd.png)
(zoom in)

**Reply by Astromations** [reply]
Instead of setting the border to 0px, set it to the background color of the status panel and sidebar so it doesn't show. I'll have to look into the loading dot later.

**Reply by Zabooby** [reply]
Works like a charm!

**Reply by Zabooby** [reply]
@Astromations Have you found out how to change the colour of the loading dot yet?
